imap/message.c

```diff
@@ -2652,7 +2652,21 @@ static void message_write_searchaddr(struct buf *buf,
 
             if (addrlist->name) {
                 tmp = charset_parse_mimeheader(addrlist->name, charset_flags);
+                /* Determine if name is an atext or quoted-string */
+                static const char *atext_specials = "!#$%&'*+-/=?^_`{|}~";
```

Comment:
What happens if there's a " (DOUBLE QUOTE) in the name?  Is that something that can happen?

Reply:
Right, it actually can! I have updated the code and tests accordingly.